### DISCUSSION TITLE
How to make kotaemon as backend.

### DISCUSSION THREAD

**Discussion by rachmatg** [Cinnamon/kotaemon#330] [General]
How to make kotaemon as backend so I can make my own frontend and access the kotaemon using API?

**Comment by taprosoft** [comment]
Hi @rachmatg sorry for the late reply. You can check this issue for using Kotaemon as API / back-end. Currently Gradio already provide built-in API for various functions of Kotaemon like changing settings, uploading files or chat endpoint. https://github.com/Cinnamon/kotaemon/issues/267

However at the moment it requires some tinkering and trial-and-error since this is not our focus in the beginning. However, we are aware that this is a common request from community and we are working on a new version that enable BE API integration more easily.

**Reply by milanmvd1** [reply]
Which is exact gradio API endpoint name for uploading files and get the upload status info.  I see many API end points in gradio client and it is confusing.  I tried at least 4-5 and none of them are working.  Any help will be appreciated.  I tried index_fn, index_fn_with_default_loaders etc.
